### ISSUE TITLE
Show the outline in explorer view only when a markdown file is active

### ISSUE THREAD

**Issue by hashhar** [neilsustc/vscode-markdown#40]
The `when` clause at https://code.visualstudio.com/docs/extensionAPI/extension-points#_contributesviews might help I guess.

**Comment by neilsustc** [comment]
I tried but I find something strange.

`package.json`:

```json
"explorer": [
    {
        "id": "mdOutline",
        "name": "Outline",
        "when": "resourceLangId == markdown"
    }
```

![when-iloveimg-compressed](https://user-images.githubusercontent.com/7588612/31072225-a90098a2-a799-11e7-9b5c-23a1194f1cb4.gif)

1. Open a folder with a Markdown file opened by default - No outline ❌
2. Switch to a `.json` file - No outline ✔
3. Switch to `.md` file - Has outline ✔
4. Switch to `.json` file - Has outline ❌

**Comment by hashhar** [comment]
That is odd. I guess it's a bug or maybe the way it is designed to work. Ping @tyriar

**Comment by Tyriar** [comment]
I think @sandy081 worked on this

**Comment by MagicPro1994** [comment]
I don't know why but it works smoothly when you turn on "markdown.extension.preview.autoShowPreviewToSide": true.

**Comment by karbassi** [comment]
Fixed in #44
